op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,278 @@
+	WebsocketServerPort int
+	// RollupBoostWsURL is the URL of the rollup boost WebSocket
+	RollupBoostWsURL string
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	wsClientMu      sync.Mutex
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	wsClients       []*websocket.Conn
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) FlashblockHandler {
+	return &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+	}
+}
+
+// Start initializes the flashblocks handler and starts the WebSocket server and rollup boost listener
+func (h *Handler) Start(ctx context.Context) error {
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener
+	if err := h.startRollupBoostListener(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+// This should only be called if the conductor is shut down
+func (h *Handler) Stop() {
+	// Close the rollup boost connection if it exists
+	if h.rollupBoostConn != nil {
+		h.rollupBoostConn.Close()
+		h.rollupBoostConn = nil
+	}
+
+	// Close all WebSocket client connections
+	h.wsClientMu.Lock()
+	for _, client := range h.wsClients {
+		client.Close()
+	}
+	h.wsClients = nil
+	h.wsClientMu.Unlock()
+
+	// Force close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("Forcibly closing WebSocket server")
+		err := h.server.Close()
+		if err != nil {
+			h.log.Error("Error closing WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.wsClientMu.Lock()
+	defer h.wsClientMu.Unlock()
+
+	if len(h.wsClients) == 0 {
+		h.log.Debug("no WebSocket clients connected, not broadcasting message")
+		return
+	}
+
+	h.log.Info("Broadcasting message to WebSocket clients", "clientCount", len(h.wsClients))
+
+	// Send message to all clients
+	for _, client := range h.wsClients {
+		err := client.WriteMessage(websocket.TextMessage, message)
+		if err != nil {
+			h.log.Error("error writing to WebSocket client", "err", err, "remote", client.RemoteAddr())
+			// Just log the error but keep the connection
+		}
+	}
+}
+
+// startRollupBoostListener initializes and starts a WebSocket client to listen for rollup boost messages
+func (h *Handler) startRollupBoostListener(ctx context.Context) error {
+	if h.cfg.RollupBoostWsURL == "" {
+		h.log.Info("rollup boost WebSocket disabled, no URL configured")
+		return nil
+	}
+
+	// Start a goroutine to maintain the rollup boost WebSocket connection
+	go func() {
+		backoff := time.Second
+		maxBackoff := 30 * time.Second
```

Comment:
You don't need to do this, just not close the connection when there's an read error is fine